pyiceberg/transforms.py

```diff
@@ -349,6 +356,13 @@ def dedup_name(self) -> str:
     def preserves_order(self) -> bool:
         return True
 
+    @abstractmethod
+    def pyarrow_transform(self, source: IcebergType) -> "Callable[[pa.Array], pa.Array]": ...
+
+    @property
+    def supports_pyarrow_transform(self) -> bool:
```

Comment:
Instead of adding a public method, how about we maintaining a private list of transform that supports pyarrow transform for a check at the beginning of `append`. For transforms that does not yet support pyarrow transform, we could throw `NotImplementedError` in `pyarrow_transform`

WDYT!

Reply:
I was following the existing convention for `can_transform` and `preserves_order` that are specified as class properties. I'm in favor of keeping it consistent with these

Reply:
Yeah I'm on the same page! To add to that - I'm not yet sure if we will be able to support the other Transform classes in `support_pyarrow_transform`. The key idea here is that we want to use native pyarrow compute functions to apply the equivalent transform without converting the values back and forth between Arrow and Python data types. I'm not yet sure if we'll be able to do the same for BucketTransform for instance, with the existing range of pyarrow compute functions.